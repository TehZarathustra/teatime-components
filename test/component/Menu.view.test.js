'use strict';

const {noop} = require('../../lib/dash');
const Menu = require('../../component/Menu');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #direction should have a value `bottom`', () => {
  expect(Menu.defaultProps.direction).toBe('bottom');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Menu
      onOutsideClick={noop}
      parentNode={noop}>
      Menu Content
    </Menu>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Menu
      className='outer'
      direction='top'
      onOutsideClick={noop}
      parentNode={noop}>
      Menu Content
    </Menu>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
