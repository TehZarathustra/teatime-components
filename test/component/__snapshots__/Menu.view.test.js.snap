// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<span
  className="menu bottom"
  role="listbox"
>
  Menu Content
</span>
`;

exports[`Renders correctly with defined props 1`] = `
<span
  className="menu top outer"
  role="listbox"
>
  Menu Content
</span>
`;
